src/util.h

```diff
@@ -68,30 +69,55 @@ inline std::string _(const char* psz)
 void SetupEnvironment();
 bool SetupNetworking();
 
+enum LogFlags : uint32_t {
+    LOG_NONE        = 0,
+    LOG_NET         = (1 <<  0),
+    LOG_TOR         = (1 <<  1),
+    LOG_MEMPOOL     = (1 <<  2),
+    LOG_HTTP        = (1 <<  3),
+    LOG_BENCH       = (1 <<  4),
+    LOG_ZMQ         = (1 <<  5),
+    LOG_DB          = (1 <<  6),
+    LOG_RPC         = (1 <<  7),
+    LOG_ESTIMATEFEE = (1 <<  8),
+    LOG_ADDRMAN     = (1 <<  9),
+    LOG_SELECTCOINS = (1 << 10),
+    LOG_REINDEX     = (1 << 11),
+    LOG_CMPCTBLOCK  = (1 << 12),
+    LOG_RAND        = (1 << 13),
+    LOG_PRUNE       = (1 << 14),
+    LOG_PROXY       = (1 << 15),
+    LOG_MEMPOOLREJ  = (1 << 16),
+    LOG_LIBEVENT    = (1 << 17),
+    LOG_COINDB      = (1 << 18),
+    LOG_QT          = (1 << 19),
+    LOG_LEVELDB     = (1 << 20),
+    LOG_ALL         = ~(uint32_t)0,
+};
+
 /** Return true if log accepts specified category */
-bool LogAcceptCategory(const char* category);
+static inline bool LogAcceptCategory(uint32_t category)
+{
+    return (logCategories.load(std::memory_order_relaxed) & category) != 0;
+}
+
+/** Returns a string with the supported log categories */
+std::string ListLogCategories();
+
+/** Return true of str parses as a log category and set the flags in f */
+bool ParseLogCategory(uint32_t *f, const std::string *str);
+
 /** Send a string to the log output */
 int LogPrintStr(const std::string &str);
 
-/** Get format string from VA_ARGS for error reporting */
-template<typename... Args> std::string FormatStringFromLogArgs(const char *fmt, const Args&... args) { return fmt; }
-
-#define LogPrintf(...) do { \
-    std::string _log_msg_; /* Unlikely name to avoid shadowing variables */ \
-    try { \
-        _log_msg_ = tfm::format(__VA_ARGS__); \
-    } catch (tinyformat::format_error &fmterr) { \
-        /* Original format string will have newline so don't add one here */ \
-        _log_msg_ = "Error \"" + std::string(fmterr.what()) + "\" while formatting log message: " + FormatStringFromLogArgs(__VA_ARGS__); \
-    } \
-    LogPrintStr(_log_msg_); \
-} while(0)
-
-#define LogPrint(category, ...) do { \
-    if (LogAcceptCategory((category))) { \
-        LogPrintf(__VA_ARGS__); \
-    } \
-} while(0)
+#define LogPrintf(...) LogPrintStr(tfm::format(__VA_ARGS__))
```

Comment:
Why have you removed the try/except handling in this macro? (I'm not saying it's wrong, I just don't understand why it was necessary before but not now).